src/core/service.c

```diff
@@ -3549,14 +3553,42 @@ static int service_dispatch_exec_io(sd_event_source *source, int fd, uint32_t ev
                                 log_unit_debug(UNIT(s), "Got EOF on exec-fd while it was disabled, ignoring.");
 
                         return 0;
+
+                case sizeof(uint8_t):
```

Comment:
uh, please don't. this just ugly. i mean, we know that pipes are never going to give us short reads, but still, yuck.

please change the read() to loop_read, and always read 64bit. And then write a value of == 0 to "disable" the concept, and > 0 to enable it/send timestamp.

i.e. determining what event happened by size rather then contents is very icky I am sure.